### PR TITLE
Make all configmaps unique in same namespace

### LINKED ISSUE #18536: Failure to start second workspace of a user on Openshift infrastructure
### Describe the bug
While I was working on a workspace start bug for Kubernetes infra, I discovered this problem.
If a user tries to start second workspace while the first one is still running, then the socond workspace failes to start.
The reason of the failure is that Che server tries to create `ca-certs` config map that already exists in the namespace (the configmap was created for the first workspace). The exception prevents second workspace from successful start.

### Che version
nightly

### Steps to reproduce
1. Deploy Eclipse Che on Openshift infrastructure (CRC possible too)
2. Allow starting several workspaces simultaneously by patching CR:
```yaml
spec:
  server:
    customCheProperties:
      CHE_LIMITS_USER_WORKSPACES_RUN_COUNT: "10"
```
3. Start a workspace
4. Without stopping previous one, try to start second workspace.

### Expected behavior
It is possible to start as many workspaces as needed.

### Runtime
Openshift

### Installation method
chectl using operator installer

### Environment
 - my computer (Linux)
 - cluster bot

### Eclipse Che Logs

Second workspace of the user start log:
```log
Error: Failed to run the workspace: "Failure executing: POST at: https://172.25.0.1/api/v1/namespaces/myusername-che/configmaps. Message: configmaps "ca-certs" already exists. Received status: Status(apiVersion=v1, code=409, details=StatusDetails(causes=[], group=null, kind=configmaps, name=ca-certs, retryAfterSeconds=null, uid=null, additionalProperties={}), kind=Status, message=configmaps "ca-certs" already exists, metadata=ListMeta(_continue=null, remainingItemCount=null, resourceVersion=null, selfLink=null, additionalProperties={}), reason=AlreadyExists, status=Failure, additionalProperties={})."
```

<details>
<summary>Che server logs</summary>

```log
2020-12-04 11:07:52,292[aceSharedPool-1]  [WARN ] [.c.w.i.k.n.KubernetesNamespace 193]  - Can't label the namespace due to lack of permissions. Grant cluster-wide permissions to `get` and `update` the `namespaces` to the `che` service account (Che operator might have already prepared a cluster role called `che-namespace-editor` for this, depending on its configuration). Alternatively, consider disabling the feature by setting `che.infra.kubernetes.namepsace.label` to `false`.
2020-12-04 11:07:55,751[aceSharedPool-1]  [WARN ] [.i.k.KubernetesInternalRuntime 257]  - Failed to start Kubernetes runtime of workspace workspacej47xvrinfedlnelf.
org.eclipse.che.workspace.infrastructure.kubernetes.KubernetesInfrastructureException: Failure executing: POST at: https://172.25.0.1/api/v1/namespaces/myusername-che/configmaps. Message: configmaps "ca-certs" already exists. Received status: Status(apiVersion=v1, code=409, details=StatusDetails(causes=[], group=null, kind=configmaps, name=ca-certs, retryAfterSeconds=null, uid=null, additionalProperties={}), kind=Status, message=configmaps "ca-certs" already exists, metadata=ListMeta(_continue=null, remainingItemCount=null, resourceVersion=null, selfLink=null, additionalProperties={}), reason=AlreadyExists, status=Failure, additionalProperties={}).
	at org.eclipse.che.workspace.infrastructure.kubernetes.namespace.KubernetesConfigsMaps.create(KubernetesConfigsMaps.java:75)
	at org.eclipse.che.workspace.infrastructure.kubernetes.wsplugins.brokerphases.DeployBroker.execute(DeployBroker.java:98)
	at org.eclipse.che.workspace.infrastructure.kubernetes.wsplugins.brokerphases.PrepareStorage.execute(PrepareStorage.java:76)
	at org.eclipse.che.workspace.infrastructure.kubernetes.wsplugins.brokerphases.ListenBrokerEvents.execute(ListenBrokerEvents.java:63)
	at org.eclipse.che.workspace.infrastructure.kubernetes.wsplugins.PluginBrokerManager.getTooling(PluginBrokerManager.java:129)
	at org.eclipse.che.workspace.infrastructure.kubernetes.wsplugins.SidecarToolingProvisioner.provision(SidecarToolingProvisioner.java:90)
	at org.eclipse.che.workspace.infrastructure.kubernetes.KubernetesInternalRuntime.provisionWorkspace(KubernetesInternalRuntime.java:288)
	at org.eclipse.che.workspace.infrastructure.kubernetes.KubernetesInternalRuntime.internalStart(KubernetesInternalRuntime.java:204)
	at org.eclipse.che.api.workspace.server.spi.InternalRuntime.start(InternalRuntime.java:141)
	at org.eclipse.che.api.workspace.server.WorkspaceRuntimes$StartRuntimeTask.run(WorkspaceRuntimes.java:969)
	at org.eclipse.che.commons.lang.concurrent.CopyThreadLocalRunnable.run(CopyThreadLocalRunnable.java:38)
	at java.base/java.util.concurrent.CompletableFuture$AsyncRun.run(Unknown Source)
	at io.micrometer.core.instrument.internal.TimedRunnable.run(TimedRunnable.java:44)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
	at org.eclipse.che.commons.observability.CountedThreadFactory.lambda$newThread$0(CountedThreadFactory.java:75)
	at java.base/java.lang.Thread.run(Unknown Source)
Caused by: io.fabric8.kubernetes.client.KubernetesClientException: Failure executing: POST at: https://172.25.0.1/api/v1/namespaces/myusername-che/configmaps. Message: configmaps "ca-certs" already exists. Received status: Status(apiVersion=v1, code=409, details=StatusDetails(causes=[], group=null, kind=configmaps, name=ca-certs, retryAfterSeconds=null, uid=null, additionalProperties={}), kind=Status, message=configmaps "ca-certs" already exists, metadata=ListMeta(_continue=null, remainingItemCount=null, resourceVersion=null, selfLink=null, additionalProperties={}), reason=AlreadyExists, status=Failure, additionalProperties={}).
	at io.fabric8.kubernetes.client.dsl.base.OperationSupport.requestFailure(OperationSupport.java:568)
	at io.fabric8.kubernetes.client.dsl.base.OperationSupport.assertResponseCode(OperationSupport.java:507)
	at io.fabric8.kubernetes.client.dsl.base.OperationSupport.handleResponse(OperationSupport.java:471)
	at io.fabric8.kubernetes.client.dsl.base.OperationSupport.handleResponse(OperationSupport.java:430)
	at io.fabric8.kubernetes.client.dsl.base.OperationSupport.handleCreate(OperationSupport.java:251)
	at io.fabric8.kubernetes.client.dsl.base.BaseOperation.handleCreate(BaseOperation.java:815)
	at io.fabric8.kubernetes.client.dsl.base.BaseOperation.create(BaseOperation.java:333)
	at org.eclipse.che.workspace.infrastructure.kubernetes.namespace.KubernetesConfigsMaps.create(KubernetesConfigsMaps.java:73)
	... 16 common frames omitted
2020-12-04 11:07:55,945[aceSharedPool-1]  [INFO ] [o.e.c.a.w.s.WorkspaceRuntimes 995]   - Workspace 'myusername:golang-opaz6' with id 'workspacej47xvrinfedlnelf' start failed
```
</details>

### Additional context
This problem is not reproducible on Kubernetes infrastructure. The reason is [this condition](https://github.com/eclipse/che/blob/286459283ad2097d7c177df829033f785a2e21ea/infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/provision/KubernetesTrustedCAProvisioner.java#L117-L119). When second workspace starts on Kubernetes then both configmaps have the same data inside and a new config map isn't added to provisioner. In case of Openshift, they are always different. The reason is the `config.openshift.io/inject-trusted-cabundle=true` label which makes Openshift add `ca-bundle.crt` field into the config map. So, they are most of the time (except cluster proxy config has additional CAs and the field is propagated to the source config map by operator) different and provisioner adds the config map to be propagated. Later, when Kubernetes objects is creating from the configs it fails [here](https://github.com/eclipse/che/blob/5e72b9edc7dadf9a9d11597ebbd2b9416e9fc992/infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/namespace/KubernetesConfigsMaps.java#L75) (the invocation is [here](https://github.com/eclipse/che/blob/14cea1b912956549cd476db6d0dc70d770c64b01/infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/wsplugins/brokerphases/DeployBroker.java#L98)) as config map with the same name already exists in the namespace.

### PR DESCRIPTION
**Pull request by skabashnyuk**
Signed-off-by: Sergii Kabashniuk <skabashniuk@redhat.com>

### What does this PR do?
Make all configmaps unique in the same namespace.
1. Removed any workspace id prefix.
2. Put `UniqueNamesProvisioner` to the end of the list of provisioners. After that, all config maps renamed and have the name `wsid.originalname`

### Screenshot/screencast of this PR
Before the change. Openshift.
Знімок екрана 2020-12-15 о 09 56 18
After the change Openshift
Знімок екрана 2020-12-15 о 10 07 04
After the change Minikube
Знімок екрана 2020-12-15 о 14 07 45
Знімок екрана 2020-12-15 о 14 07 35

### What issues does this PR fix or reference?
Fixes https://github.com/eclipse/che/issues/18536

### How to test this PR?
1. Deploy che server image `quay.io/skabashn/che-server:che18536'
2. Create two workspaces. Start them.
3. All config maps have to have different names.

### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)

- [x] [The Eclipse Contributor Agreement is valid](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-eclipse-contributor-agreement-is-valid)
- [x] [Code produced is complete](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-produced-is-complete)
- [x] [Code builds without errors](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-builds-without-errors)
- [x] [Tests are covering the bugfix](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#tests-are-covering-the-bugfix)
- [x] [The repository devfile is up to date and works](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-repository-devfile-is-up-to-date-and-works)
- [x] [Sections `What issues does this PR fix or reference` and `How to test this PR` completed](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#sections-what-issues-does-this-pr-fix-or-reference-and-how-to-test-this-pr-completed)
- [x] [Relevant user documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [x] [Relevant contributing documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [x] [CI/CD changes implemented, documented and communicated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#cicd-changes-implemented-documented-and-communicated)

### Reviewers

Reviewers, please comment how you tested the PR when approving it.

### CODE REVIEW

File: infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/provision/KubernetesTrustedCAProvisioner.java [hunk 1]
```diff
@@ -113,26 +112,21 @@ public void provision(KubernetesEnvironment k8sEnv, RuntimeIdentity runtimeID)
...
-    }
+
+    k8sEnv
+        .getConfigMaps()
+        .put(
+            configMapName,
+            new ConfigMapBuilder()
+                .withMetadata(
+                    new ObjectMetaBuilder()
+                        .withName(configMapName)
```
Comment (mmorhun) [inline infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/provision/KubernetesTrustedCAProvisioner.java:120]: I think description of `che.infra.kubernetes.trusted_ca.dest_configmap` property should be updated

**Review by mshaposhnik — APPROVED** [review]

File: infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/provision/KubernetesTrustedCAProvisioner.java (hunk 1 above)
Comment (skabashnyuk) [reply infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/provision/KubernetesTrustedCAProvisioner.java:120]: done

**Review by amisevsk — APPROVED** [review]

**Review by mmorhun — APPROVED** [review]
Tested on CRC, works for me

File: infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/provision/KubernetesTrustedCAProvisioner.java [hunk 2]
```diff
@@ -113,26 +112,21 @@ public void provision(KubernetesEnvironment k8sEnv, RuntimeIdentity runtimeID)
     }
 
     KubernetesNamespace namespace = namespaceFactory.getOrCreate(runtimeID);
-    Optional<ConfigMap> existing = namespace.configMaps().get(configMapName);
```
Comment (sparkoo) [inline infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/provision/KubernetesTrustedCAProvisioner.java:116]: why this check is not needed anymore?

File: infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/provision/SshKeysProvisioner.java [hunk 3]
```diff
@@ -250,7 +250,7 @@ private void doProvisionVcsSshConfig(
       sshConfigData.append(buildConfig(sshPair.getName()));
     }
 
-    String sshConfigMapName = wsId + SSH_CONFIG_MAP_NAME_SUFFIX;
+    String sshConfigMapName = SSH_CONFIG_MAP_NAME;
```
Comment (sparkoo) [inline infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/provision/SshKeysProvisioner.java]: do we need this variable anymore? can't we use the constant directly?

**Review by sparkoo — APPROVED** [review]

File: infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/provision/KubernetesTrustedCAProvisioner.java (hunk 2 above)
Comment (skabashnyuk) [reply infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/provision/KubernetesTrustedCAProvisioner.java:116]: We do have exactly one configmap per workspace that is created just before the workspace creating. We are not trying to update the content of existed configmap and use it for all workspaces.

**Pull request merged**